src/core/declarativewebcontainer.cpp

```diff
@@ -851,17 +852,17 @@ void DeclarativeWebContainer::updateLoading()
     emit loadingChanged();
 }
 
-void DeclarativeWebContainer::onFirstPaint(int, int)
-{
-    updateActiveTabRendered();
-}
-
 void DeclarativeWebContainer::updateActiveTabRendered()
 {
+    if (m_webPage && !m_webPage->completed()) {
```

Comment:
If we get the compositingFinished() signal before m_webPage->completed() is true, does this mean that we will never setActiveTabRendered(true), or does the compositingFinished() signal get emitted once per frame?  I assume the latter, but wanted to double check.